plugins/grid-bookmark/src/GridBookmarkWidget/model.ts

```diff
@@ -150,20 +173,43 @@ export default function f(_pluginManager: PluginManager) {
       ) {
         bookmark.correspondingObj.setLabel(label)
       },
+      updateBookmarkHighlight(
+        bookmark: IExtendedLabeledRegionModel,
+        color: string,
+      ) {
+        bookmark.correspondingObj.setHighlight(color)
+      },
+      updateBulkBookmarkHighlights(color: string) {
+        self.selectedBookmarks.forEach(bookmark =>
+          this.updateBookmarkHighlight(bookmark, color),
+        )
+      },
       setSelectedBookmarks(bookmarks: IExtendedLabeledRegionModel[]) {
         self.selectedBookmarks = bookmarks
       },
       setBookmarkedRegions(regions: IMSTArray<typeof LabeledRegionModel>) {
         self.bookmarks = cast(regions)
       },
+      setHighlightToggle(toggle: boolean) {
+        const { views } = getSession(self)
+        ;(views as IExtendedLGV[]).forEach(view => {
+          view.toggleShowBookmarkHighlights?.(toggle)
+        })
+      },
+      setLabelToggle(toggle: boolean) {
+        const { views } = getSession(self)
+        ;(views as IExtendedLGV[]).forEach(view => {
+          view.toggleShowBookmarkLabels?.(toggle)
+        })
+      },
     }))
     .actions(self => ({
       clearAllBookmarks() {
-        for (const bookmark of self.bookmarks) {
-          if (self.validAssemblies.has(bookmark.assemblyName)) {
-            self.bookmarks.remove(bookmark)
-          }
-        }
+        self.setBookmarkedRegions(
```

Comment:
any reason for this change?

Reply:
iterating through the array of bookmarks and then removing them was causing issues when clearing all bookmarks -- not all bookmarks were cleared. this function is only used when you attempt to delete bookmarks with nothing selected ('clear' operation)

filtering out bookmarks not present in the session's assemblies then casting the property in setBookmarkedRegions seems to work better.

Reply:
that is a good bugfix, but it calls into question what was wrong with the previous approach. does the observable-array::remove function not work properly here, or is something not properly observing it

Reply:
(i can reproduce the issue where 'delete'/clear all was not working before)